app/download/views.py

```diff
@@ -77,6 +120,49 @@ def download_document_b64(service_id, document_id):
     except ValueError:
         abort(404)
 
+    try:
+        check_scan_verdict(service_id, document_id, sending_method)
+    except (MaliciousContentError, SuspiciousContentError) as e:
+        current_app.logger.info(
+            "Malicious content detected, refused to download document: {}".format(e),
+            extra={
+                "service_id": service_id,
+                "document_id": document_id,
+            },
+        )
+        return jsonify(error=str(e)), MALICIOUS_CONTENT_ERROR_CODE
+    except ScanInProgressError as e:
+        age_seconds = scan_files_document_store.get_object_age_seconds(
+            service_id, document_id, sending_method
+        )
+        if age_seconds > SCAN_TIMEOUT_SECONDS:
+            current_app.logger.info(
+                "Scan timed out for document: {}".format(e),
+                extra={
+                    "service_id": service_id,
+                    "document_id": document_id,
+                },
+            )
+            return jsonify(scan_verdict="scan_timed_out"), 200
+
+        current_app.logger.info(
+            "Scan in progress, refused to download document: {}".format(e),
+            extra={
+                "service_id": service_id,
+                "document_id": document_id,
+            },
+        )
+        return jsonify(error=str(e)), SCAN_IN_PROGRESS_ERROR_CODE
```

Comment:
```suggestion
    # at this point the email with the "link" type attachment has been sent
    # return the document to the user in case the scan timed out
    continue
```

If we return a 200 here, the user will not get the document they are requesting. Instead we can just continue on to the code below where the response with the file is sent.